Consul.Test/ClientTest.cs

```diff
@@ -94,6 +94,31 @@ public async Task Client_SetQueryOptions()
             Assert.Equal("1m40s", request.Params["wait"]);
         }
 
+        [Fact]
+        public async Task Client_UseCache()
```

Comment:
This test, doesn't actually demonstrate what is the valid way of using cache, could you implement a test that does it?